### DISCUSSION TITLE
Log4J Version used by Gitblit

### DISCUSSION THREAD

**Discussion by Micha0123** [gitblit/gitblit#1472] [Q&A]
Hi, 
is there a way to change the log4j version gitblit uses? it-sec wants me to disable all outdated log4j-versions

**Comment by flaix** [comment]
You should actually be able to replace the used log4j with a current version of reload4j. Gitblit comes with its dependencies JARs in a directory, so that it is possible to easily replace them.

In your installation find the `ext` folder.
Then replace the three files
`log4j-1.2.17.jar`
`slf4j-api-1.7.29.jar`
`slf4j-log4j12-1.7.29.jar`

with the new files
`reload4j-1.2.25.jar` ([here](https://repo1.maven.org/maven2/ch/qos/reload4j/reload4j/1.2.25/reload4j-1.2.25.jar))
`slf4j-api-1.7.36.jar` ([here](https://repo1.maven.org/maven2/org/slf4j/slf4j-api/1.7.36/))
`slf4j-reload4j-1.7.36.jar` ([here](https://repo1.maven.org/maven2/org/slf4j/slf4j-reload4j/1.7.36/))

Restart your server and make sure that everything still works properly as expected.

**Comment by flaix** [comment]
Since Gitblit version 1.10.0 has switched from log4j1 to reload4j, I am setting this topic as closed.
